test/gtest/common/mem_buffer.cc

```diff
@@ -229,11 +229,20 @@ uint64_t mem_buffer::pat(uint64_t prev) {
 }
 
 mem_buffer::mem_buffer(size_t size, ucs_memory_type_t mem_type) :
-    m_mem_type(mem_type), m_ptr(allocate(size, mem_type)), m_size(size) {
+    m_mem_type(mem_type), m_ptr(allocate(size, mem_type)), m_size(size),
+    m_allocated(1) {
+}
+
+mem_buffer::mem_buffer(void *ptr, size_t size, ucs_memory_type_t mem_type) :
+    m_mem_type(mem_type), m_ptr(ptr ? ptr : allocate(size, mem_type)),
```

Comment:
need to use `ptr != NULL`: 
```
m_ptr((ptr != NULL) ? ptr : allocate(size, mem_type))
```

Reply:
hmmmm... both versions are allowed, but ok, updated

Reply:
we use `NULL` in UCX code, my point to align with it :)